test/testutil.py

```diff
@@ -147,3 +149,270 @@ def create_filesystem_customizations(rootfs: str):
+    def lookaside_conf_file(self, file: str) -> None:
+        self._lookaside_conf_file = file
+
+    @property
+    def lookaside_conf(self) -> str:
+        return self._lookaside_conf
+
+    @lookaside_conf.setter
+    def lookaside_conf(self, conf: str) -> None:
+        self._lookaside_conf = conf
+
+
+def ensure_registry():
+    registry_container_name = subprocess.run([
+        "podman", "ps", "-a", "--filter", "name=registry", "--format", "{{.Names}}"
+    ], check=True, capture_output=True).stdout.decode("utf-8").strip()
+
+    if registry_container_name != "registry":
+        subprocess.run([
+            "podman", "run", "-d",
+            "-p", "5000:5000",
+            "--restart", "always",
+            "--name", "registry",
+            "registry:2"
+        ], check=True, capture_output=True)
+
+    registry_container_state = subprocess.run([
+        "podman", "ps", "-a", "--filter", "name=registry", "--format", "{{.State}}"
+    ], check=True, capture_output=True).stdout.decode("utf-8").strip()
+
+    if registry_container_state in ("paused", "exited"):
+        subprocess.run([
+            "podman", "start", "registry"
+        ], check=True, capture_output=True)
+
+
+def get_signed_container_ref(local_registry: str, container_ref: str):
+    container_ref_path = container_ref[container_ref.index('/'):]
+    return f"{local_registry}{container_ref_path}"
+
+
+def sign_container_image(gpg_conf: GPGConf, registry_conf: RegistryConf, container_ref):
+    gpg_gen_key(gpg_conf)
+    ensure_registry()
+    local_registry = registry_conf.local_registry
+    policy_file = registry_conf.policy_file
+    lookaside_conf_file = registry_conf.lookaside_conf_file
+    lookaside_conf = registry_conf.lookaside_conf
+    pub_key_file = gpg_conf.pub_key_file
+    registry_policy = {
+        "default": [{"type": "insecureAcceptAnything"}],
+        "transports": {
+            "docker": {
+                f"{local_registry}": [
+                    {
+                        "type": "signedBy",
+                        "keyType": "GPGKeys",
+                        "keyPath": f"{pub_key_file}"
+                    }
+                ]
+            },
+            "docker-daemon": {
+                "": [{"type": "insecureAcceptAnything"}]
+            }
+        }
+    }
+    with open(policy_file, mode="w", encoding="utf-8") as f:
+        f.write(json.dumps(registry_policy))
+
+    with open(lookaside_conf_file, mode="w", encoding="utf-8") as f:
+        f.write(lookaside_conf)
+
+    signed_container_ref = get_signed_container_ref(local_registry, container_ref)
+    system_lookaside_conf_file = os.path.join(
+        "/etc/containers/registries.d",
+        os.path.basename(lookaside_conf_file)
+    )
+    # We need to temporarily configure system's lookaside for skopeo to
+    # create the sigstore dir.
+    shutil.copy(lookaside_conf_file, system_lookaside_conf_file)
+    subprocess.run([
+        "skopeo", "copy",
```

Comment:
Offline I tried to use skopeo copy to sign the container image, but it looks like it is time consuming since it also need copy container image.  I try this way as per https://github.com/containers/podman/blob/main/docs/tutorials/image_signing.md. Here firstly retag image to localhost, then use sudo -E GNUPGHOME=$HOME/.gnupg \
    podman push \
    --tls-verify=false \
    --sign-by sgrunert@suse.com \
    localhost:5000/alpine.
It is straightforward, and much time saving

Reply:
I am not sure I understand, doing it with podman would be exactly the same as doing it with skopeo. Doing it with podman requires multiple commands: first pull, then tag and then sign+push the container to the new registry. Skopeo does all of that in one single command and, I haven't tested, but it should take moreless the same amount of time. Are you saying that there is a big difference? How much?

Reply:
My tried actions follow as:
1)firstly pull image to local host
2)podman tag image to local registry
3)podman sign local image and push back to local registry.
Since all operations almost happen on container on local registry, it happen very fast. Since it aim to  testing purpose only, so i don't need to push back to remote registry,such as quay.io. 

From your analysis, it looks like the process is almost same.
Maybe I missed something in last attempt using skopeo copy